src/neuroglancer/segmentation_user_layer.ts

```diff
@@ -493,6 +513,19 @@ export class SegmentationUserLayer extends Base {
               for (const renderLayer of graphRenderLayers) {
                 loadedSubsource.addRenderLayer(renderLayer);
               }
+              const graphTab = segmentationGraph.tab;
```

Comment:
One thing that I am a bit concerned about is how it will work to have this tab selected, or dragged out as a separate panel, given that it is added only once the data source is loaded, rather than when the layer is first created.

Have you tested that?

Reply:
I just tested pinning the graph tab and toggling the graph source off and back on. The pinned tab stays, the tool UI inside the tab hides but the list of annotations remains. It reverts back to normal when toggling the source back on.

Reply:
What about if you reload the browser while the graph tab is selected or dragged out as a separate panel?

Reply:
It disappears/reverts back to the source tab.

Reply:
I think to get better behavior with this tab a slightly different approach may be required --- I think for the layer tabs we may need a concept of a "hidden tab", which is shown only when selected --- if not selected, no tab is shown.

Then when a graph subsource with a graph tab is activated, the tab is marked non-hidden.

Additionally the graph connection would not supply the actual top-level tab, but instead something that would included as the contents of the tab, e.g. via DependentViewWidget.

Reply:
What about making the graphConnection property on the segmentationLayer a watchable value? That way I can create a tab class that listens to that and replaces the graph tab content and toggle it's own visibility. Rather than accessing the tab from the segmentationLayer.

Also currently I had the tab method on the source rather than connection but it shouldn't be a problem moving it to the connection.

Here is my current WIP https://github.com/seung-lab/neuroglancer/commit/3d6c0726972557da4add5fa926209ade3795393f

Reply:
Yes I think that change makes sense.

To be clear I was suggesting that the `DependentViewWidget` stuff be done by segmentation_user_layer, not by graphene-specifically --- graphene would supply its tabContents, and the segmentation layer would ensure it is visible when that graph source is activated.

Reply:
I converted layer.graphConnection to a WatchableValue but I ended up using `layer.displayState.segmentationGroupState.value.graph` instead. I forgot that the graph was accessible through that. Should I revert the change to graphConnection or does it seem useful to keep?

https://github.com/google/neuroglancer/pull/395/commits/f25632e3ed592dfd9bb908357e43dca5ecd459c3